pages/blog/_slug.vue

```diff
@@ -154,18 +154,22 @@ export default {
   content: {
     liveEdit: false,
   },
-  async asyncData({ params }) {
+  async asyncData({ params, error }) {
     const response = await axios.get(`${process.env.strapiUrl}/api/blogs`, {
       params: {
         'filters[slug][$eq]': params.slug,
         populate: '*',
       },
     })
     const blog = response.data.data[0]
+    if (!blog?.id) {
```

Comment:
This filter applies automatically right? ,so why blog will not present.

Reply:
So when we enter a wrong slug it was not giving 404 error.. Instead it was giving some internal errors like `id of undefined`

Reply:
So why we enter the wrong slug, Do you mean slug is entered by user.

Reply:
Maybe by the user or some wrong links that might be present in the blogs.. So previously we changed the existing slugs of some of the feature pages and forgot to change those links in the blogs.. Now I have fixed all the blogs but in future we might have this kind of change.. So added the fallback for that.

Reply:
So is there any way through which if user to access any route which is not found so it will automatically gives not found.
See if strapi or next provides some configuration to handle that.

Reply:
So if strapi returns a 404 error then it would directly show that error... but as we are filtering we don't get a 404 error instead we get an empty array. 
Ideally, this should be handled by the generate method in the nuxt.config.js but not sure why we are getting client-side errors.